test/e2e/managed_resource_vap_test.go

```diff
@@ -48,19 +53,132 @@ var managedByLabelMap = map[string]string{
 	managedByLabel: managedByLabelValue,
 }
 
+// Need this because Entry() evaluates parameters at definition time, not at runtime.
+// Without this, the client value sent to Entry() would always be nil.
+func getUserClient() client.Client {
+	return hubCluster.ImpersonateKubeClient
+}
+
+func getAksServiceClient() client.Client {
+	return hubCluster.SystemMastersClient
+}
```

Comment:
nit: group all helper functions together (in the end)

Reply:
done